### ISSUE TITLE
do not show blacklisted tracking sites

### ISSUE THREAD

**Issue by harridu** [RequestPolicy/requestpolicy#368]
It would be very nice if I could permanently blacklist tracking sites. They should not be shown in the RequestPolicy menu to be enabled again.

**Comment by pfrancois** [comment]
or when we choose some site they never accept request even if we ask to accept all.

**Comment by aghast** [comment]
NoScript offers this feature, and I think it's a good one. There are sites, like the web trackers, that I will never allow, and which don't offer any benefit to have in the menu (other than telling me that yet another site is trying to track me).

Please do create a "Always deny" or "Never allow" or (to copy from NS) "Mark ___ as untrusted" option.

**Comment by myrdd** [comment]
I agree.

I think that this could be done even easier – this is in distinguishing between permanently and temporarily blocked/allowed requests. So in fact there would be six lists:
- permanent allow-rule
- permanent block-rule
- temporary allow-rule
- temporary block-rule
- without rule – block or allow according to the default policy (RP Settings → Manage Policies → Default Policy)
- Mixed rules

Some of them may be hidden, like in NoScript as suggested by aghast.

In my opinion there is no need for a special button like „Put on blacklist“ as this is handled internally already as „permanent“

**Comment by vsviridov** [comment]
+1
Going through the list of things I know I never want enabled each time is slightly annoying.

**Comment by higuita** [comment]
Agree, being a noscript user, i really miss a blacklist.
when i check if a site is useful or not, i take the decision on that time... but later, on another site i dont recall already have analyzed that third-party site and i will check it again.
with a blacklist (or some flag to distinguish from the other blocked sites), i could simply ignore that site and continue to browser.

**Comment by Ranger2000** [comment]
I always would like to see who is blocking what site.
And for tracking-sites I would prefer to use the Addon http://www.ghostery.com/
Here, this problem has been solved in a clever way.

**Comment by higuita** [comment]
ok, no need to really remove the blacklist sites from the list, just mark then grey on the blocked list. this way we can see new sites from already checked and rejected sites.

**Comment by kiwidude** [comment]
This would be very useful indeed, because then blacklisted sites do not waste UI space (and my time).
Two implementation details would be important:
- Blacklisted sites do not cause the flag to go red, thus telling me that there are no more sites I might wish to allow. Currently the flag is always red (and therefore carrying no information) because just about every site uses things like useless-analytics.com that I never ever have any use for.
- The black list should be accessible and viewable from the usual menu (e.g. the flag's popup menu), with the destinations used by the currently viewed page at the top and marked somehow, like a different colour or bold face. This allows to verify whether something ended up on the blacklist that perhaps shouldn't be on it; there is nothing quite as annoying than site that can't be made to work with the current settings... IOW don't block something behind the scenes for which the reason is not discoverable by the user.

**Comment by c33s** [comment]
see https://github.com/RequestPolicy/requestpolicy/issues/6

**Comment by RobotManager** [comment]
Agreed 100%, suggesting this feature is half the reason I registered.  We really do need a way to hide those domains we don't ever intend to enable.  On some sites the list is insane, making it very difficult to manage at times and hard to find new domains.
